test/gtest/ucs/test_profile.cc

```diff
@@ -115,59 +179,136 @@ void test_profile::test_locations(ucs_profile_location_t *locations,
+                EXPECT_EQ((uintptr_t)&test_request, rec->param64);
+            }
+        }
+
+        ptr = records + hdr->num_records;
     }
 
-    std::string data = p.read();
-    ucs_profile_header_t *hdr = reinterpret_cast<ucs_profile_header_t*>(&data[0]);
-    test_header(hdr, UCS_BIT(UCS_PROFILE_MODE_LOG));
-
-    EXPECT_EQ(12u, hdr->num_locations);
-    ucs_profile_location_t *locations = reinterpret_cast<ucs_profile_location_t*>(hdr + 1);
-    test_locations(locations, hdr->num_locations, 0);
-
-    EXPECT_EQ(12 * ITER, (int)hdr->num_records);
-    ucs_profile_record_t *records = reinterpret_cast<ucs_profile_record_t*>(locations +
-                                                                            hdr->num_locations);
-    uint64_t prev_ts = records[0].timestamp;
-    for (uint64_t i = 0; i < hdr->num_records; ++i) {
-        ucs_profile_record_t *rec = &records[i];
-        EXPECT_GE(rec->location, 0u);
-        EXPECT_LT(rec->location, 12u);
-        EXPECT_GE(rec->timestamp, prev_ts);
-        prev_ts = rec->timestamp;
-        ucs_profile_location_t *loc = &locations[rec->location];
-        if ((loc->type == UCS_PROFILE_TYPE_REQUEST_NEW) ||
-            (loc->type == UCS_PROFILE_TYPE_REQUEST_EVENT) ||
-            (loc->type == UCS_PROFILE_TYPE_REQUEST_FREE))
-        {
-            EXPECT_EQ((uintptr_t)&test_request, rec->param64);
+    EXPECT_EQ(&data[data.size()], ptr) << data.size();
+}
+
+UCS_TEST_P(test_profile, accum) {
+    do_test(UCS_BIT(UCS_PROFILE_MODE_ACCUM), "accum");
+}
+
+UCS_TEST_P(test_profile, log) {
+    do_test(UCS_BIT(UCS_PROFILE_MODE_LOG), "log");
+}
+
+UCS_TEST_P(test_profile, log_accum) {
+    do_test(UCS_BIT(UCS_PROFILE_MODE_LOG) | UCS_BIT(UCS_PROFILE_MODE_ACCUM),
+            "log,accum");
+}
+
+INSTANTIATE_TEST_CASE_P(st, test_profile, ::testing::Values(1));
+
+class test_profile_perf : public test_profile {
+};
+
+UCS_TEST_SKIP_COND_P(test_profile_perf, overhead, RUNNING_ON_VALGRIND) {
+
+    const double EXP_OVERHEAD_NSEC = 50.0;
+    const int ITERS                = 100;
+    const int COUNT                = 1000000;
+    double overhead_nsec           = 0.0;
+
+    for (int retry = 0; retry < (ucs::perf_retry_count + 1); ++retry) {
+        ucs_time_t  time_profile_on  = 0;
+        ucs_time_t  time_profile_off = 0;
+
+        for (int i = 0; i < ITERS; ++i) {
+            ucs_time_t t;
+
+            t = ucs_get_time();
+            for (volatile int j = 0; j < COUNT;) {
+                ++j;
+            }
+            if (i > 2) {
+                time_profile_off += ucs_get_time() - t;
+            }
+
+            t = ucs_get_time();
+            for (volatile int j = 0; j < COUNT;) {
+                UCS_PROFILE_CODE("test") {
+                    ++j;
+                }
+            }
+            if (i > 2) {
+                time_profile_on += ucs_get_time() - t;
+            }
+        }
+
+        overhead_nsec = ucs_time_to_nsec(time_profile_on - time_profile_off) / COUNT;
```

Comment:
If I got it in the right way, it should be
`COUNT` -> `ITERS` and we shouldn't consider warmup iterations (i.e. `ITERS - 2`
```
overhead_nsec = ucs_time_to_nsec(time_profile_on - time_profile_off) / (ITERS - 2);
```

Or if we measure overhead per `++j`, it should be `(ITERS - 2) * COUNT`

BTW, what about to define `WARMUP_ITERS = 2`?